yjit/src/codegen.rs

```diff
@@ -1245,6 +1276,36 @@ mod tests {
         assert_eq!(tmp_type_top, Type::Fixnum);
     }
 
+    #[test]
+    fn test_putself() {
+        let mut context = Context::new();
+        let mut cb = CodeBlock::new();
+        let mut ocb = OutlinedCb::wrap(CodeBlock::new());
+        let mut jit = JITState::new();
+
+        let status = gen_putself(&mut jit, &mut context, &mut cb, &mut ocb);
```

Comment:
You don't need to do that for this pull request, but I think it would make sense to add a function that generates the codegen objects and add that to the `test` module, eg:

let (jit, ctx, cb, ocb) = setup_codegen();

Just because, the specifics of the setup code could change and it would save a lot of boilerplate.

Reply:
Yup. I had been planning to do that, then realised I had a PR open that would need to mess with it again :-) But yeah, it's on my list.